lib/src/header/all_day_events.dart

```diff
@@ -42,15 +44,25 @@ class AllDayEvents<E extends Event> extends StatelessWidget {
               events = events.where((e) => e.intersectsInterval(visibleDates));
 
               return ValueListenableBuilder(
-                valueListenable: controller.scrollControllers.pageListenable,
-                builder: (context, page, __) =>
-                    _buildEventLayout(context, events, page),
-              );
+                  valueListenable: controller.scrollControllers.pageListenable,
+                  builder: (context, page, __) => GestureDetector(
+                        behavior: HitTestBehavior.translucent,
+                        onTapUp: (details) {
+                          final tappedCell = details.localPosition.dx / (constraints.widthConstraints().maxWidth / controller.visibleRange.visibleDays);
+                          final date = LocalDate.fromEpochDay(page.floor() + tappedCell.toInt());
+                          final dateAndTime = DateTime(date.year, date.monthOfYear, date.dayOfYear + 2);
```

Comment:
Similar to the part-day calculation, I'd prefer only using types from <kbd>time_machine</kbd> for consistency. You can add two days with `date + Period(days: 2)`. I'm still curious about that two days, though. One day might come from a different timezone that results in a wrong day at 1:00 or 23:00 – this should be fixed by not converting to Dart's native `DateTime`. But two days… Can you please check if that number changes when viewing a different number of days (setting a different `VisibleRange` in the `Timetable` widget)?